Sources/ComposableArchitecture/Instrumentation.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+
+/// Interface to enable tracking/instrumenting the activity within TCA as ``Actions`` are sent into ``Store``s and
+/// ``ViewStores``, ``Reducers`` are executed, and ``Effects`` are observed.
+///
+/// The way the library will call the closures provided is identical to the way that the ``Actions`` and ``Effects`` are
+/// handled internally. That means that there is likely to be ``Instrumentation.ViewStore`` `will|did` pairs contained
+/// within the bounds of an ``Instrumentation.Store`` `will|did` pair. For example: Consider sending a simple ``Action``
+/// into a ``ViewStore`` that does not produce any synchronous ``Effects`` to be generated by a ``Reducer``, and the
+/// ``ViewStore`` is scoped off a parent ``Store``s state:
+///   1. An ``Action`` is passed to ``ViewStore.send``
```

Comment:
formatting nit: unless you're gonna refer to the numbers elsewhere (which I don't see a reason to here), dropping them would make this more readable (and make future changes more readable as you don't need to renumber everything :)).
In fact might be even more readable as a nested diagram, with nesting showing the will/did pairs. 
Up to you.